binding.c

```diff
@@ -474,18 +768,54 @@ NAPI_METHOD(udx_napi_socket_init) {
   napi_create_reference(env, argv[6], 1, &(self->realloc_message));
 
   int err = udx_socket_init((udx_t *) udx, socket);
-  if (err < 0) UDX_NAPI_THROW(err)
+  if (err < 0) {
+    napi_throw_error(env, uv_err_name(err), uv_strerror(err));
+    return NULL;
+  }
 
   return NULL;
 }
 
-NAPI_METHOD(udx_napi_socket_bind) {
-  NAPI_ARGV(5)
-  NAPI_ARGV_BUFFER_CAST(udx_socket_t *, self, 0)
-  NAPI_ARGV_UINT32(port, 1)
-  NAPI_ARGV_UTF8(ip, INET6_ADDRSTRLEN, 2)
-  NAPI_ARGV_UINT32(family, 3)
-  NAPI_ARGV_UINT32(flags, 4)
+napi_value
+udx_napi_socket_bind (napi_env env, napi_callback_info info) {
+  napi_value argv[5];
+  size_t argc = 5;
+  if (napi_get_cb_info(env, info, &argc, argv, NULL, NULL) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_cb_info(env, info, &argc, argv, NULL, NULL) failed!");
+    return NULL;
+  }
+
+  udx_socket_t *self;
+  size_t self_len;
+  if (napi_get_buffer_info(env, argv[0], (void **) &self, &self_len) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_buffer_info(env, argv[0], (void **) &self, &self_len) failed!");
+    return NULL;
+  }
+
+  uint32_t port;
+  if (napi_get_value_uint32(env, argv[1], &port) != napi_ok) {
+    napi_throw_error(env, "EINVAL", "Expected unsigned number");
+    return NULL;
+  }
+
+  char ip[INET6_ADDRSTRLEN];
+  size_t ip_len;
+  if (napi_get_value_string_utf8(env, argv[2], (char *) &ip, INET6_ADDRSTRLEN, &ip_len) != napi_ok) {
+    napi_throw_error(env, "EINVAL", "Expected string");
+    return NULL;
+  }
+
+  uint32_t family;
+  if (napi_get_value_uint32(env, argv[3], &family) != napi_ok) {
+    napi_throw_error(env, "EINVAL", "Expected unsigned number");
+    return NULL;
+  }
+
+  uint32_t flags;
+  if (napi_get_value_uint32(env, argv[4], &flags) != napi_ok) {
+    napi_throw_error(env, "EINVAL", "Expected unsigned number");
```

Comment:
same as above, just ignore the errors from the napi_ calls, the js only calls it safely